test/e2e/rte.go

```diff
@@ -104,6 +107,25 @@ var _ = ginkgo.Describe("[RTE][InfraConsuming] Resource topology exporter", func
 			gomega.Expect(reason).To(gomega.Equal(nrtupdater.RTEUpdateReactive), "update reason error: expected %q got %q", nrtupdater.RTEUpdateReactive, reason)
 		})
 	})
+
+	ginkgo.Context("with the binary available", func() {
+		ginkgo.It("it should show the correct version", func() {
+			out1, err := execCmd([]string{
+				filepath.Join(utils.BinariesPath, "resource-topology-exporter"),
+				"--version",
+			})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			out2, err := execCmd([]string{
+				"git",
+				"describe", "--tags", "--dirty", "--always",
+			})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			gomega.Expect(out1).To(gomega.Equal(out2))
```

Comment:
In general, test should not be aware of the implementation, as much as possible. In this case, I'd just check the output of the version is a valid version, possibly using the semantic versioning.

Reply:
but call ```version.Get()``` from the test will return an undefined version since it's injected only at build time.

Reply:
If it returns undefined, how it can be equal to the output of `git describe ... `?

Reply:
Anyway, something along these lines:
```
import (
        ...
	"github.com/hashicorp/go-version"
        ...
)

var _ = ginkgo.Describe("rte version", func() {
	ginkgo.Context("with the tool available", func() {
		ginkgo.It("it should show the correct version", func() {
			cmdline := []string{
				filepath.Join(binariesPath, "resource-topology-exporter"),
				"version",
			}
			fmt.Fprintf(ginkgo.GinkgoWriter, "running: %v\n", cmdline)

			cmd := exec.Command(cmdline[0], cmdline[1:]...)
			cmd.Stderr = ginkgo.GinkgoWriter

			out, err := cmd.Output()
			gomega.Expect(err).ToNot(gomega.HaveOccurred())

			text := strings.TrimSpace(string(out))
			fmt.Fprintf(ginkgo.GinkgoWriter, "reported version: %q\n", text)
			gomega.Expect(text).ToNot(gomega.BeEmpty())
			_, err = version.NewVersion(text)
			gomega.Expect(err).ToNot(gomega.HaveOccurred())
		})
	})
})
```

Reply:
> If it returns undefined, how it can be equal to the output of `git describe ... `?

Because we're not calling ```version.Get()``` from the code directly, but invoking the binary itself which has the version already injected